temporalio/temporal
# Fix DeleteHistoryBranch

**What changed?**
1) Move serialization / deserialization out of store layer and up to manager layer for DeleteHistoryBranch.
2) Fix a bug where the old code count current branch's ancestor in ref counting. This will prevent some ancestor branches that is only used by current branch to be deleted.
3) Added new unit test to cover the fixed bug.


Clear layer separation, fix bug.



**How did you test it?**
Unit test. Also test manually with admin tool with steps:
1) Start a workflow
2) Reset workflow using tctl
3) Using admin tool to show branch info of the new workflow (it shows tree_id, branch_id and ancestor branch_id).
4) Using admin tool to delete the original workflow and the new workflow.
5) Verify in cqlsh directly that the history_node does not contains any rows for the tree_id. 
Note: same steps with old code shows some left over events in history_nodes after both execution are deleted.


**Potential risks**
High.


**Is hotfix candidate?**
No.
